### PR TITLE
drivers: sensor: tmp1075: Add tmp1075 sensor driver and sample

### PR DESCRIPTION
**Pull request by Paplewski**
Simple PR adding support for tmp1075 temperature sensor and sample app.

### CODE REVIEW

File: drivers/sensor/ti/tmp1075/Kconfig [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	help
+	  Enable the driver for Texas Instruments TMP1075 High-Accuracy Digital
+	  Temperature Sensors.
+
+if TMP1075
+
+config TMP1075_ALERT_INTERRUPTS
+	bool "Allow interrupts to service over and under temp alerts"
+	help
+	  This will set up interrupts to service under and over temp alerts
```
Comment (treto) [inline drivers/sensor/ti/tmp1075/Kconfig:18]: id proly just use a full name for over and under temperature alerts (follow naming from datasheet)

File: drivers/sensor/ti/tmp1075/tmp1075.h [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	const struct gpio_dt_spec alert_gpio;
+	uint8_t cr;
+	uint8_t cf;
+	bool alert_pol: 1;
+	bool one_shot: 1;
+	bool interrupt_mode: 1;
+	bool shutdown_mode: 1;
+};
+
+int tmp1075_trigger_set(const struct device *dev, const struct sensor_trigger *trig,
```
Comment (treto) [inline drivers/sensor/ti/tmp1075/tmp1075.h:69]: don't we need docstrings for api functions?

File: drivers/sensor/ti/tmp1075/tmp1075_trigger.c [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+LOG_MODULE_DECLARE(TMP1075, CONFIG_SENSOR_LOG_LEVEL);
+
+/*
+ * @brief GPIO alert line interrupt calback
+ * @param gpio - not used
+ * @param cb - callback structure for interrupt handler
+ * @param pins - not used
+ */
+void tmp1075_trigger_handle_alert(const struct device *gpio, struct gpio_callback *cb,
```
Comment (treto) [inline drivers/sensor/ti/tmp1075/tmp1075_trigger.c:21]: i'm not certain it's not overcomplicating things that these two functions are here and not in tmp1075.c, it's a simple device driver in the end, there are no #ifdefs in .h to force this trigger support to be necessary

File: drivers/sensor/ti/tmp1075/tmp1075_trigger.c [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+
+#include "tmp1075.h"
+
+LOG_MODULE_DECLARE(TMP1075, CONFIG_SENSOR_LOG_LEVEL);
+
+/*
+ * @brief GPIO alert line interrupt calback
```
Comment (treto) [inline drivers/sensor/ti/tmp1075/tmp1075_trigger.c]: typo

File: drivers/sensor/ti/tmp1075/tmp1075_trigger.c [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+ * @param handler - pointer to custom callback handler which the user would like to use
+ * @return 0 if ok - -ENOTSUP in case of err
+ */
+int tmp1075_trigger_set(const struct device *dev, const struct sensor_trigger *trig,
+			sensor_trigger_handler_t handler)
+{
+	struct tmp1075_data *drv_data = dev->data;
+
+	if (trig->type == SENSOR_TRIG_THRESHOLD) {
+		drv_data->temp_alert_handler = handler;
```
Comment (treto) [inline drivers/sensor/ti/tmp1075/tmp1075_trigger.c:48]: is sensor_trigger_handler_t  void? if not you are not handling status in line 27

File: drivers/sensor/ti/tmp1075/tmp1075_trigger.c [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+ * @brief callback implementation for setting the custom trigger handler in the userspace
+ * @param dev - sensor device struct pointer
+ * @param trig - trigger struct pointer to be set up
+ * @param handler - pointer to custom callback handler which the user would like to use
+ * @return 0 if ok - -ENOTSUP in case of err
+ */
+int tmp1075_trigger_set(const struct device *dev, const struct sensor_trigger *trig,
+			sensor_trigger_handler_t handler)
+{
+	struct tmp1075_data *drv_data = dev->data;
```
Comment (treto) [inline drivers/sensor/ti/tmp1075/tmp1075_trigger.c:45]: not checking for nulls

File: dts/bindings/sensor/ti,tmp1075.yaml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  TMP1075 Digital Temperature Sensor.  See more info at
+  https://www.ti.com/product/TMP1075
+
+compatible: "ti,tmp1075"
+
+include: [sensor-device.yaml, i2c-device.yaml]
+
+properties:
+  conversion-rate:
+    description: Conversion rate in ms
```
Comment (treto) [inline dts/bindings/sensor/ti,tmp1075.yaml]: if 275 is 27.5ms, then to be precise this is Conversion rate in 100us (or 0.1ms)

File: dts/bindings/sensor/ti,tmp1075.yaml [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    type: int
+    default: 220
```
Comment (treto) [inline dts/bindings/sensor/ti,tmp1075.yaml:16]: i'd add this comment that this is # 22.0ms and remove it from line 18

File: dts/bindings/sensor/ti,tmp1075.yaml [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+description: |
+  TMP1075 Digital Temperature Sensor.  See more info at
+  https://www.ti.com/product/TMP1075
+
+compatible: "ti,tmp1075"
+
+include: [sensor-device.yaml, i2c-device.yaml]
+
+properties:
```
Comment (treto) [inline dts/bindings/sensor/ti,tmp1075.yaml:12]: incosistent "." placement in sentences, sometimes you do, sometimes you dont

File: dts/bindings/sensor/ti,tmp1075.yaml [hunk 10]
```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2024 Arrow Electronics.
```
Comment (treto) [inline dts/bindings/sensor/ti,tmp1075.yaml:1]: not sure this boilerplate works with zephyr?

File: dts/bindings/sensor/ti,tmp1075.yaml [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+  alert-gpios:
+    type: phandle-array
+    description: |
+      Identifies the ALERT signal, which is active-low open drain when
+      produced by the sensor.
+  one-shot-conversion:
+    description: One-shot conversion mode.
+    type: boolean
+  consecutive-fault-measurements:
+    description: Consecutive fault measurements to trigger the alert.
```
Comment (treto) [inline dts/bindings/sensor/ti,tmp1075.yaml]: I'd rephrase it to:
Number of consecutive measured faults that will trigger the alert.

File: dts/bindings/sensor/ti,tmp1075.yaml [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    type: int
+    default: 4
+    enum:
+      - 1
+      - 2
+      - 3
+      - 4
+  alert-pin-active-high:
+    description: Polarity of the alert pin
```
Comment (treto) [inline dts/bindings/sensor/ti,tmp1075.yaml]: alert-pin-active-high - doesn't it indicate polarity? shouldn't this be like alert-pin-polarity instead?

File: dts/bindings/sensor/ti,tmp1075.yaml [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+    type: boolean
+  interrupt-mode:
+    description: Selects the function of the ALERT pin
```
Comment (treto) [inline dts/bindings/sensor/ti,tmp1075.yaml]: mode and boolean seem kinda odd, this sounds more like an enum; if there are 2 classes then explain them in description.

File: dts/bindings/sensor/ti,tmp1075.yaml [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+    type: boolean
+  shutdown-mode:
+    description: Sets the device in shutdown mode to conserve power
```
Comment (treto) [inline dts/bindings/sensor/ti,tmp1075.yaml]: what does it mean for device boot time behavior? that a user must manually enable or power it up or something? I'd put a note like that

File: dts/bindings/sensor/ti,tmp1075.yaml [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+    type: boolean
+  lower-threshold:
+    description: Lower threshold for alert interrupt Expressed in degrees C.
```
Comment (treto) [inline dts/bindings/sensor/ti,tmp1075.yaml]: typo, i guess you are missing . before Expressed. Also, I'd add where this is inclusively 28 or not. Do we need 27.9 to trigger or 28?

File: samples/sensor/tmp1075/README.rst [hunk 16]
```diff
@@ -0,0 +1,18 @@
+.. _tmp1075_sample:
+
+TMP1075 sample
+#############
+
+Description
+***********
+
+This sample writes the temperature to the console once every 1 second.
+If TMP1075_ALERT_INTERRUPTS is enabled the it is possible to use interrupts
```
Comment (treto) [inline samples/sensor/tmp1075/README.rst]: typo, "then it is"

File: drivers/sensor/ti/tmp1075/tmp1075.h (hunk 2 above)
Comment (Paplewski) [reply drivers/sensor/ti/tmp1075/tmp1075.h:69]: this is internal zephyr sensor driver function used in one place - this is implementation of sensor_trigger_set not used by the user's API

File: drivers/sensor/ti/tmp1075/tmp1075_trigger.c (hunk 3 above)
Comment (Paplewski) [reply drivers/sensor/ti/tmp1075/tmp1075_trigger.c:21]: i guess this is convention for the sensor to separate trigger file, i copied the structure from other tmp sensor

File: dts/bindings/sensor/ti,tmp1075.yaml (hunk 10 above)
Comment (Paplewski) [reply dts/bindings/sensor/ti,tmp1075.yaml:1]: for yaml `#` works, but for c/h files we have to use multiline style comment

File: dts/bindings/sensor/ti,tmp1075.yaml (hunk 12 above)
Comment (Paplewski) [reply dts/bindings/sensor/ti,tmp1075.yaml]: yes it does, the value can be either true or false so `alert-pin-active-high: true` is active-high `alert-pin-active-high: false` is active-low

File: drivers/sensor/ti/tmp1075/Kconfig (hunk 1 above)
Comment (Paplewski) [reply drivers/sensor/ti/tmp1075/Kconfig:18]: I don't get what you mean

File: drivers/sensor/ti/tmp1075/tmp1075_trigger.c (hunk 5 above)
Comment (Paplewski) [reply drivers/sensor/ti/tmp1075/tmp1075_trigger.c:48]: i dont get the question.
in case the sensor trigger handler is not set then it is void and don't reach line 27

if it is set (by this function) then its nonvoid and line 27 executed the handler set in this line

File: dts/bindings/sensor/ti,tmp1075.yaml (hunk 8 above)
Comment (Paplewski) [reply dts/bindings/sensor/ti,tmp1075.yaml:16]: default value is 220ms, ill add more comments if its not clear enough

File: dts/bindings/sensor/ti,tmp1075.yaml (hunk 13 above)
Comment (Paplewski) [reply dts/bindings/sensor/ti,tmp1075.yaml]: its either default or interrupt mode - it should be clear when you have sensor datasheet handy, otherwise all these parameters are like some kind of magic switches

File: dts/bindings/sensor/ti,tmp1075.yaml (hunk 15 above)
Comment (Paplewski) [reply dts/bindings/sensor/ti,tmp1075.yaml]: i don't know :c

File: drivers/sensor/ti/tmp1075/tmp1075.h (hunk 2 above)
Comment (Paplewski) [reply drivers/sensor/ti/tmp1075/tmp1075.h:69]: the docstring is in the `.c` file

File: drivers/sensor/ti/tmp1075/tmp1075_trigger.c (hunk 4 above)
Comment (Paplewski) [reply drivers/sensor/ti/tmp1075/tmp1075_trigger.c]: fixed

File: drivers/sensor/ti/tmp1075/tmp1075_trigger.c (hunk 6 above)
Comment (Paplewski) [reply drivers/sensor/ti/tmp1075/tmp1075_trigger.c:45]: if it was null it would not reach here, but ok its better to be on the safe side. check added

File: dts/bindings/sensor/ti,tmp1075.yaml (hunk 9 above)
Comment (Paplewski) [reply dts/bindings/sensor/ti,tmp1075.yaml:12]: added periods.

File: dts/bindings/sensor/ti,tmp1075.yaml (hunk 7 above)
Comment (Paplewski) [reply dts/bindings/sensor/ti,tmp1075.yaml]: these are numbers from datasheet, ill leave it as it is - added more comments

File: dts/bindings/sensor/ti,tmp1075.yaml (hunk 11 above)
Comment (Paplewski) [reply dts/bindings/sensor/ti,tmp1075.yaml]: done

File: dts/bindings/sensor/ti,tmp1075.yaml (hunk 14 above)
Comment (Paplewski) [reply dts/bindings/sensor/ti,tmp1075.yaml]: i don't know what it does - everything is explained in the datasheet. I just enabled the user the switch for convenience

File: samples/sensor/tmp1075/README.rst (hunk 16 above)
Comment (Paplewski) [reply samples/sensor/tmp1075/README.rst]: fixed
